### PR TITLE
#FN-30: User should be able to Search

### PR DESCRIPTION
**Pull request by Charlesntwari**
Trello: https://trello.com/c/F22E0TCF/30-users-should-be-able-to-search-for-products

# PR Description

 Buyer Search for Products

## Description of tasks that were expected to be completed

List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style

- [x] search used product name.
- [x] search by using category.
- [x] search by using price range.

## Description of tasks that were expected to be completed and that are not yet completed

- [x] none

## How this may be tested?

Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

1. run `yarn i` to install all required dependencies
2. run `yarn run dev`
3. go on products page
4. then search product

## Screenshots (If appropriate/applicable)

Not necessary

## Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by Illustre13 — COMMENTED** [review]
Well Done, but kindly first resolve conflicts on this PR

**Review by Niyobuzima — CHANGES_REQUESTED** [review]
Good job, kindly resolve the conflicts on this PR

**Review by Honore123 — APPROVED** [review]

**Review by dushimeemma — CHANGES_REQUESTED** [review]
Kindly read the task description and implement all the required scenarios @Charlesntwari.

**Review by Niyobuzima — APPROVED** [review]

File: src/components/HomePage.tsx [hunk 1]
```diff
@@ -7,7 +8,9 @@ const HomePage: React.FC = () => {
       <div className="bg-[#bae1f4] hero-bg-md:bg-hero-bg bg-cover hero-bg:bg-center min-h-[85vh] flex flex-col justify-center items-start hero-bg-md:pl-24 hero-bg-md:items-start items-center">
         <p className="font-bold hero-bg-md:text-4xl text-3xl text-[#003D29] hero-bg-md:max-w-xs hero-bg-md:mb-4 mb-3 max-w-[80%] text-center hero-bg-md:text-left">Shooping And Department Store</p>
         <p className="hero-bg-md:max-w-xs mb-4 max-w-[80%] hero-bg-sm:max-w-[50%] text-center hero-bg-md:text-left text-lg hero-bg-md:mb-6 mb-8">Get whatever you need from the best sellers and the best price</p>
-        <Button label="Get Started" style="bg-[#003D29] text-white hero-bg-md:px-10 hero-bg-md:w-fit w-[40%] py-2 rounded-3xl" />
+        <Link to="/products/all">
```
Comment (Illustre13) [inline src/components/HomePage.tsx]: I think this should a `/products` instead `/products/all`

**Review by Illustre13 — CHANGES_REQUESTED** [review]
Also works on the commit message

File: src/components/HomePage.tsx (hunk 1 above)
Comment (Charlesntwari) [reply src/components/HomePage.tsx]: done

**Review by Illustre13 — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]

**Pull request merged**